checker-qual/src/main/java/org/checkerframework/checker/calledmethods/qual/EnsuresCalledMethodsOnException.java

```diff
@@ -0,0 +1,83 @@
+package org.checkerframework.checker.calledmethods.qual;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.InheritedAnnotation;
+
+/**
+ * Indicates that the method, if it terminates by throwing an exception, always invokes the given
+ * methods on the given expressions. This annotation is repeatable, which means that users can write
+ * more than one instance of it on the same method (users should NOT manually write an
+ * {@code @EnsuresCalledMethodsOnException.List} annotation, which the checker will create from
+ * multiple copies of this annotation automatically).
+ *
+ * <p>Consider the following method:
+ *
+ * <pre>
+ * &#64;EnsuresCalledMethodsOnException(value = "#1", methods = "m")
+ * public void callM(T t) { ... }
+ * </pre>
+ *
+ * <p>This method guarantees that {@code t.m()} is always called before the method throws an
+ * exception.
```

Comment:
This covers `RuntimeException` since `RuntimeException extends Exception`. I will add a comment clarifying that this does not cover `Error` or other throwables.